e2e/internal/e2e/imageverify.go

```diff
@@ -0,0 +1,406 @@
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+// ImageVerify checks for an image integrity
+func ImageVerify(t *testing.T, cmdPath string, imagePath string, labels bool, runDisabled bool) {
+	type testSpec struct {
+		name          string
+		execArgs      []string
+		expectSuccess bool
+	}
+	tests := []testSpec{
+		{"False", []string{"false"}, false},
+		{"RunScript", []string{"test", "-f", "/.singularity.d/runscript"}, true},
+		{"OneBase", []string{"test", "-f", "/.singularity.d/env/01-base.sh"}, true},
+		{"ActionsShell", []string{"test", "-f", "/.singularity.d/actions/shell"}, true},
+		{"ActionsExec", []string{"test", "-f", "/.singularity.d/actions/exec"}, true},
+		{"ActionsRun", []string{"test", "-f", "/.singularity.d/actions/run"}, true},
+		{"Environment", []string{"test", "-L", "/environment"}, true},
+		{"Singularity", []string{"test", "-L", "/singularity"}, true},
+	}
+	if labels && runDisabled { // TODO
```

Comment:
What do you need to TODO here?

Reply:
Looks like this is no longer needed. Seems like an artifact of when singularity was still incomplete but adam was trying to write some basic tests